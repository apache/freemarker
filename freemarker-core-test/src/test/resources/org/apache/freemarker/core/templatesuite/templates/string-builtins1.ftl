<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
FreeMarker: Encoding string built-in tests

<#assign x = r'  dieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x ${"kigyo"?upperCase}  '>
  
capFirst:   ${x?capFirst};
uncapFirst: ${x?uncapFirst};
uncapFirst: ${"Blah"?uncapFirst}
capitalize: ${x?capitalize};
html:       ${x?html};
length:     ${x?length};
lowerCase:  ${x?lowerCase};
rtf:        ${x?rtf};
trim:       ${x?trim};
trim2:      ${"foo bar"?trim}
trim3:      ${" foo bar"?trim}
trim4:      ${"foo bar "?trim}
upperCase:  ${x?upperCase};
xml:        ${x?xml};
xhtml:      ${"\"Blah's is > 1 & < 2\""?xhtml}
<@assertEquals actual="'"?html expected="&#39;" />
<@assertEquals actual="'"?xhtml expected="&#39;" />
<@assertEquals actual="'"?xml expected="&apos;" />
<#-- ?substring: -->
<@assertEquals actual="ab"?substring(0) expected="ab" />
<@assertEquals actual="ab"?substring(1) expected="b" />
<@assertEquals actual="ab"?substring(2) expected="" />
<@assertFails message="at least 0">${"ab"?substring(-1)}</@><#t>
<@assertFails message="greater than the length of the string">${"ab"?substring(3)}</@><#t>
<@assertEquals actual="ab"?substring(0, 0) expected="" />
<@assertEquals actual="ab"?substring(0, 1) expected="a" />
<@assertEquals actual="ab"?substring(0, 2) expected="ab" />
<@assertFails message="at least 0">${"ab"?substring(0, -1)}</@><#t>
<@assertFails message="greater than the length of the string">${"ab"?substring(0, 3)}</@><#t>
<@assertEquals actual="ab"?substring(1, 1) expected="" />
<@assertEquals actual="ab"?substring(1, 2) expected="b" />
<@assertFails message="at least 0">${"ab"?substring(1, -1)}</@><#t>
<@assertFails message="greater than the length of the string">${"ab"?substring(1, 3)}</@><#t>
<@assertFails message="shouldn't be greater than the end index">${"ab"?substring(1, 0)}</@><#t>

word_list:
<#global words = x?wordList>
<#list words as w>- ${w}
</#list>

<#global canufeelitbabe = x?interpret>
interpret: <@canufeelitbabe></@>
<#setting locale="es_ES">number: ${"-123.45"?number + 1.1}
${"1.5e3"?number?c}
${"0005"?number?c}
${"+0"?number?c}
${"-0"?number?c}
${"NaN"?number?isNan?c}
${("INF"?number?isInfinite && "INF"?number > 0)?c}
${("-INF"?number?isInfinite && "-INF"?number < 0)?c}
${("Infinity"?number?isInfinite && "Infinity"?number > 0)?c}
${("-Infinity"?number?isInfinite && "-Infinity"?number < 0)?c}

${"org.apache.freemarker.core.templatesuite.models.NewTestModel"?new()}
${"org.apache.freemarker.core.templatesuite.models.NewTestModel"?new(1)}
${"org.apache.freemarker.core.templatesuite.models.NewTestModel"?new("xxx")}
${"org.apache.freemarker.core.templatesuite.models.NewTestModel"?new("xxx", "yyy")}

<#assign x = "In the beginning, God created the Heavens and The Earth.">

${x?replace("the", "The Sacred, Holy", "i")} <#-- case insensitive replacement -->
${x?replace("the", "the very", "f")} <#-- replace only the first one -->
${x?replace("", "|")} <#-- replace empry string -->
${x?replace("", "|", "f")} <#-- replace first empty string -->

${x?replace("the H[a-z]+", "the sky", "r")} <#-- regexp replacement -->

<#if x?matches(".*Heav..s.*")>matches<#else>Really?</#if>

<#list x?matches("(the) ([a-z]+)", "i") as match>
  ${match}
  ${match?groups[1]} sacred ${match?groups[2]}
</#list>  

<#assign matches = x?matches("In the ([a-z]+), God created (.*)")>
${matches?groups[0]}
${matches?groups[1]}
${matches?groups[2]}

<#assign x="foo, bar;baz,     foobar">
<#list x?split("[,;] ?", "r") as word>
   ${word}
</#list>


<#assign a = "foo", b="bar", c="(a+b)?upperCase">
${c?eval}

[${"a"?jString}] = [a]
[${"a\\'x'\nb"?jString}] = [a\\'x'\nb]
[${"\x0001\x001A "?jString}] = [\u0001\u001a ]

[${"a"?jsString}] = [a]
[${"a\\'x'\nb"?jsString}] = [a\\\'x\'\nb]
[${"\x0001\x001A "?jsString}] = [\x01\x1A ]
[${"<![CDATA["?jsString}] = [\x3C![CDATA[]
[${"]]>"?jsString}] = []]\>]

[${"a"?jsonString}] = [a]
[${"a\\'x'\nb"?jsonString}] = [a\\'x'\nb]
[${"\x0001\x001A "?jsonString}] = [\u0001\u001A ]
[${"\n\r\t\f\b\""?jsonString}] = [\n\r\t\f\b\"]
[${"/"?jsonString}] = [\/]
[${"a/b"?jsonString}] = [a/b]
[${"</script>"?jsonString}] = [<\/script>]
[${"<![CDATA["?jsonString}] = [\u003C![CDATA[]
[${"]]>"?jsonString}] = []]\u003E]
