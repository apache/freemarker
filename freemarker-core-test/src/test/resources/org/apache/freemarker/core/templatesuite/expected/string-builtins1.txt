/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
FreeMarker: Encoding string built-in tests

  
capFirst:     DieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x ${"kigyo"?upperCase}  ;
uncapFirst:   dieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x ${"kigyo"?upperCase}  ;
uncapFirst: blah
capitalize:   Diebugsdie! * Vazzze 123456 --cdc-- --<<--@ X ${"kigyo"?uppercase}  ;
html:         dieBugsDie! * vazzZE 123456 --cdc-- --&lt;&lt;--@ x ${&quot;kigyo&quot;?upperCase}  ;
length:     70;
lowerCase:    diebugsdie! * vazzze 123456 --cdc-- --<<--@ x ${"kigyo"?uppercase}  ;
rtf:          dieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x $\{"kigyo"?upperCase\}  ;
trim:       dieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x ${"kigyo"?upperCase};
trim2:      foo bar
trim3:      foo bar
trim4:      foo bar
upperCase:    DIEBUGSDIE! * VAZZZE 123456 --CDC-- --<<--@ X ${"KIGYO"?UPPERCASE}  ;
xml:          dieBugsDie! * vazzZE 123456 --cdc-- --&lt;&lt;--@ x ${&quot;kigyo&quot;?upperCase}  ;
xhtml:      &quot;Blah&#39;s is &gt; 1 &amp; &lt; 2&quot;

word_list:
- dieBugsDie!
- *
- vazzZE
- 123456
- --cdc--
- --<<--@
- x
- ${"kigyo"?upperCase}

interpret:   dieBugsDie! * vazzZE 123456 --cdc-- --<<--@ x KIGYO  
number: -122,35
1500
5
0
0
true
true
true
true
true

default constructor
1
xxx
xxx:yyy


In The Sacred, Holy beginning, God created The Sacred, Holy Heavens and The Sacred, Holy Earth. 
In the very beginning, God created the Heavens and The Earth. 
|I|n| |t|h|e| |b|e|g|i|n|n|i|n|g|,| |G|o|d| |c|r|e|a|t|e|d| |t|h|e| |H|e|a|v|e|n|s| |a|n|d| |T|h|e| |E|a|r|t|h|.| 
|In the beginning, God created the Heavens and The Earth. 

In the beginning, God created the sky and The Earth. 

matches

  the beginning
  the sacred beginning
  the Heavens
  the sacred Heavens
  The Earth
  The sacred Earth

In the beginning, God created the Heavens and The Earth.
beginning
the Heavens and The Earth.

   foo
   bar
   baz
       foobar


FOOBAR

[a] = [a]
[a\\'x'\nb] = [a\\'x'\nb]
[\u0001\u001a ] = [\u0001\u001a ]

[a] = [a]
[a\\\'x\'\nb] = [a\\\'x\'\nb]
[\x01\x1A ] = [\x01\x1A ]
[\x3C![CDATA[] = [\x3C![CDATA[]
[]]\>] = []]\>]

[a] = [a]
[a\\'x'\nb] = [a\\'x'\nb]
[\u0001\u001A ] = [\u0001\u001A ]
[\n\r\t\f\b\"] = [\n\r\t\f\b\"]
[\/] = [\/]
[a/b] = [a/b]
[<\/script>] = [<\/script>]
[\u003C![CDATA[] = [\u003C![CDATA[]
[]]\u003E] = []]\u003E]
